packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -4505,6 +4505,21 @@ void main() {
       final Text hintTextWidget = tester.widget(hintTextFinder);
       expect(hintTextWidget.style!.overflow, decoration.hintStyle!.overflow);
     });
+
+    testWidgets('hint should be ignore its height when input text', (WidgetTester tester) async {
+      final String hintText = 'hint text' * 20;
+      final InputDecoration decoration = InputDecoration(
+        hintText: hintText,
+        maintainHintHeight: false,
+      );
+
+      await tester.pumpWidget(
+        buildInputDecorator(
+          decoration: decoration,
+        ),
+      );
+      expect(find.text(hintText), findsNothing);
+    });
```

Comment:
Also, add another test without specifying `maintainHintHeight` to test the default behavior.